src/main/kotlin/com/varabyte/kobweb/intellij/colors/KobwebColorProvider.kt

```diff
@@ -0,0 +1,222 @@
+    val nextElement = when (element) {
+        is KtDotQualifiedExpression -> element.selectorExpression
+
+        is KtNameReferenceExpression -> when {
+            element.parent is KtCallExpression -> element.parent // Element is name of a function
+            else -> element.locateSource()
+        }
+
+        is KtProperty -> when {
+            element.hasInitializer() -> element.initializer
+            element.hasCustomGetter() -> element.getter
+            else -> null
+        }
+
+        is KtPropertyAccessor -> element.bodyExpression
+
+        is KtCallExpression -> (null).apply {
+            val callee = ((element.calleeExpression as? KtNameReferenceExpression)?.locateSource() as? KtNamedFunction)
+                ?: return@apply
+
+            when {
+                callee.isKobwebColorFunction("rgb(r: Int, g: Int, b: Int)") -> run {
+                    val args = element.valueArguments.evaluateArguments<Int>(3) ?: return@run
+                    return safeRgbColor(args[0], args[1], args[2])
+                }
+
+                callee.isKobwebColorFunction("rgb(value: Int)") -> run {
+                    val args = element.valueArguments.evaluateArguments<Int>(1) ?: return@run
+                    return safeRgbColor(args[0])
+                }
+
+                callee.isKobwebColorFunction("rgba(value: Int)", "rgba(value: Long)") -> run {
+                    val args = element.valueArguments.run {
+                        evaluateArguments<Int>(1) ?: evaluateArguments<Long, Int>(1) { it.toInt() }
+                    } ?: return@run
+                    return safeRgbColor(args[0] shr Byte.SIZE_BITS)
+                }
+
+                callee.isKobwebColorFunction("argb(value: Int)", "argb(value: Long)") -> run {
+                    val args = element.valueArguments.run {
+                        evaluateArguments<Int>(1) ?: evaluateArguments<Long, Int>(1) { it.toInt() }
+                    } ?: return@run
+                    return safeRgbColor(args[0] and 0x00_FF_FF_FF)
+                }
+
+                callee.isKobwebColorFunction("hsl(h: Float, s: Float, l: Float)") -> run {
+                    val args = element.valueArguments.evaluateArguments<Float>(3) ?: return@run
+                    return safeHslColor(args[0], args[1], args[2])
+                }
+
+                callee.isKobwebColorFunction("hsla(h: Float, s: Float, l: Float, a: Float)") -> run {
+                    val args = element.valueArguments.evaluateArguments<Float>(4) ?: return@run
+                    return safeHslColor(args[0], args[1], args[2])
+                }
+            }
+        }
+
+        else -> null
+    }
+
+    return if (currentDepth <= MAX_SEARCH_DEPTH) {
+        nextElement?.let { traceColor(it, currentDepth + 1) }
+    } else null
+}
+
+/**
+ * Pretty much emulates what CTRL-clicking on a reference would do
+ */
+private fun KtSimpleNameExpression.locateSource(): PsiElement? {
```

Comment:
I feel like this HAS to be something provided by the IntelliJ framework. I could be wrong. Did you already look for this in their codebase?

The action would be called "Go to declaration" (keyboard shortcut is CTRL+B/CMD+B if you want to look it up in your keymap)

Reply:
I did and there is a GotoDeclarationAction class but the methods it called didn't work for me. This was kind of my last resort. If you find something better please tell me

Reply:
Sounds good. Let's add some comments here about why we're doing our own thing, and then I'll approve.

Reply:
Found something way better